### PR TITLE
Support for Bulgarian holidays

### PR DESCRIPTION
**Pull request by dyanakiev**
# Contributing a new country ?

* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/spatie/holidays/pulls) for the same country?

1. Create a new class in the Countries directory. It should extend the Country class.
2. Add a test for the new country in the tests directory.
3. Run the tests so a snapshot gets created.
4. Verify the result in the newly created snapshot is correct.

### CODE REVIEW

File: src/Countries/Bulgaria.php [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        ];
+    }
+
+    /**
+     * Calculate the date of Easter using the Computus algorithm with Carbon
+     * @param int $year
+     * @return CarbonImmutable
+     * @throws \RuntimeException if the calculated date is invalid
+     */
+    protected function calculateEasterDate($year): CarbonImmutable
```
Comment (Nielsvanpach) [inline src/Countries/Bulgaria.php:55]: Is this an orthodox calculation? We already have one in Country.php. Does that one work too?

Comment (dyanakiev) [reply src/Countries/Bulgaria.php]: I totally didn't see that. Yes it works for Bulgaria too.

Comment (Nielsvanpach) [reply src/Countries/Bulgaria.php]: I think you used the wrong one.

Comment (dyanakiev) [reply src/Countries/Bulgaria.php]: Yep you're right.. Its correct for 2024 `5 May 2024`, but for 2025 it should be `20 April 2025`. I'm not sure which calculation to use..

Comment (dyanakiev) [reply src/Countries/Bulgaria.php]: Ok using the `easter_date` function directly actually resolves the issue.

Comment (dyanakiev) [reply src/Countries/Bulgaria.php]: Ok, nevermind.. 🤣  It's now wrong for 2024, im confused..

Comment (Nielsvanpach) [reply src/Countries/Bulgaria.php]: Country.php has a easter() and orthodoxEaster() function. I suggest to use one of those. 

If the orthodox calculation gives a false result, feel free to update it, in order to make it work. Do verify if the results for NorthMacedonia is still correct.

Comment (dyanakiev) [reply src/Countries/Bulgaria.php]: Neither functions give proper result, i will investigate more.

Comment (lowv-developer) [reply src/Countries/Bulgaria.php]: The problem is that this function in Country.php is wrong calculating the OrthodoxEastern. Please check Greece #114 to fix the problem. @Nielsvanpach needs to update orthodoxEaster() 

> Country.php has a easter() and orthodoxEaster() function.

Comment (Nielsvanpach) [reply src/Countries/Bulgaria.php]: It's still the wrong function you're calling. You should use orthodoxEaster()

File: src/Countries/Bulgaria.php [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            'Съединението на България' => '09-06',
+            'Ден на независимостта на България' => '09-22',
+            'Коледа ден 1' => '12-24',
+            'Коледа ден 2' => '12-25',
+            'Коледа ден 3' => '12-26',
+        ], $this->variableHolidays($year));
+    }
+
+    /**
+     * @param int $year
```
Comment (Nielsvanpach) [inline src/Countries/Bulgaria.php:36]:
```suggestion
```

File: src/Countries/Bulgaria.php [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+{
+    public function countryCode(): string
+    {
+        return 'bg';
+    }
+
+    /**
+     * @param int $year
+     * @return array<string, string>
+     */
```
Comment (Nielsvanpach) [inline src/Countries/Bulgaria.php:17]:
```suggestion
```

File: src/Countries/Bulgaria.php [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    protected function variableHolidays(int $year): array
+    {
+        $easter = CarbonImmutable::createFromTimestamp(easter_date($year))
+            ->setTimezone('Europe/Sofia');
```
Comment (Nielsvanpach) [inline src/Countries/Bulgaria.php:42]:
```suggestion
        $easter = $this->orthodoxEaster($year);
```

File: tests/.pest/snapshots/Countries/BulgariaTest/it_can_calculate_bulgarian_holidays.snap [hunk 5]
```diff
@@ -0,0 +1,46 @@
+[
+    {
+        "name": "\u041d\u043e\u0432\u0430 \u0433\u043e\u0434\u0438\u043d\u0430",
```
Comment (Nielsvanpach) [inline tests/.pest/snapshots/Countries/BulgariaTest/it_can_calculate_bulgarian_holidays.snap:3]: If needed, you ca also add a translation file to the `lang` directory

**Review by Nielsvanpach — CHANGES_REQUESTED** [review]

**Comment by Nielsvanpach** [comment]
Could you have a look at the failing test?

**Pull request closed without merging**